### PR TITLE
fixes #1206 - Adds new endpoint DELETE /v2/tasks to delete multiple task...

### LINKED ISSUE #1206: Add new endpoint to kill a given list of tasks
Prerequisite for https://github.com/mesosphere/marathon/issues/1207

### PR DESCRIPTION
**Pull request by drexin**
...s at once

### CODE REVIEW

**Comment by drexin** [comment]
@ConnorDoyle or @aquamatthias Please review

File: src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala [hunk 1]
```diff
@@ -17,8 +17,10 @@ class TaskIdUtil {
     appId.safePath + taskDelimiter + uuidGenerator.generate()
   }
 
-  def appId(taskId: TaskID): PathId = {
-    taskId.getValue match {
+  def appId(taskId: TaskID): PathId = appId(taskId.getValue)
+
+  def appId(taskId: String): PathId = {
+    taskId match {
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala]: Please handle the failing case explicitly, since that means you have to handle a MatchError

File: src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala [hunk 2]
```diff
@@ -56,6 +59,36 @@ class TasksResource @Inject() (
...
   ))
 
+  @DELETE
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Timed
+  def killTasks(
+    @QueryParam("scale")@DefaultValue("false") scale: Boolean,
+    body: Array[Byte]): Response = {
+    val taskIds = Json.parse(body).as[Seq[String]]
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala]: Why do we use play json here? Is jersey not capable of deserializing a Seq[String]?

File: src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala [hunk 3]
```diff
@@ -56,6 +59,36 @@ class TasksResource @Inject() (
...
+
+    val groupedTasks = taskIds.flatMap { taskId =>
+      val appId = try {
+        taskIdUtil.appId(taskId)
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala]: If that would return Either or Option we would not have to use try/catch here, which is rather clumsy.

**Comment by aquamatthias** [comment]
A question and a comment. Besides that it looks good.

File: src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala (hunk 2 above)
Comment (drexin) [inline src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala]: The goal is to remove Jackson in the future, so everything that's being added should use play-json instead.

File: src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala (hunk 3 above)
Comment (drexin) [inline src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala]: True, but should be done in a separate PR, because it requires lots of changes in places unrelated to this PR.

File: src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala (hunk 1 above)
Comment (drexin) [inline src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala]: See above comment. Has to be done in another PR.

File: src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala (hunk 2 above)
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala]: Makes sense then.

File: src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala (hunk 3 above)
Comment (drexin) [inline src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala]: Issue here: https://github.com/mesosphere/marathon/issues/1225
